Source/Projections/Builder/ClassProjectionMethod.cs

```diff
@@ -0,0 +1,96 @@
+        /// <param name="eventType">The <see cref="EventType" /> of the event.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector"/> for selecting the key.</param>
+        public ClassProjectionMethod(TaskResultProjectionMethodSignature<TProjection> method, EventType eventType, KeySelector keySelector)
+        {
+            _method = method;
+            KeySelector = keySelector;
+            _eventType = eventType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClassProjectionMethod{TEventHandler}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="SyncProjectionMethodSignature{TEvent, TProjection}" /> method to invoke.</param>
+        /// <param name="eventType">The <see cref="EventType" /> of the event.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector"/> for selecting the key.</param>
+        public ClassProjectionMethod(SyncProjectionMethodSignature<TProjection> method, EventType eventType, KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, object @event, ProjectionContext context) =>
+                {
+                    method(instanceAndReadModel, @event, context);
+                    return ProjectionResultType.Replace;
+                },
+                eventType,
+                keySelector)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ClassProjectionMethod{TEventHandler}"/> class.
+        /// </summary>
+        /// <param name="method">The <see cref="SyncResultProjectionMethodSignature{TEvent, TProjection}" /> method to invoke.</param>
+        /// <param name="eventType">The <see cref="EventType" /> of the event.</param>
+        /// <param name="keySelector">The <see cref="Projections.KeySelector"/> for selecting the key.</param>
+        public ClassProjectionMethod(SyncResultProjectionMethodSignature<TProjection> method, EventType eventType, KeySelector keySelector)
+            : this(
+                (TProjection instanceAndReadModel, object @event, ProjectionContext context)
+                    => Task.FromResult(method(instanceAndReadModel, @event, context)),
+                eventType,
+                keySelector)
+        {
+        }
+
+        /// <inheritdoc/>
+        public KeySelector KeySelector { get; }
+
+        /// <inheritdoc/>
+        public EventType GetEventType(IEventTypes eventTypes)
+            => _eventType;
+
+        /// <inheritdoc/>
+        public async Task<Try<ProjectionResult<TProjection>>> TryOn(TProjection readModel, object @event, ProjectionContext context)
+        {
+            var resultType = await _method(readModel, @event, context).TryTask().ConfigureAwait(false);
+            return resultType.Exception ?? new Try<ProjectionResult<TProjection>>(readModel);
```

Comment:
I might be missing something, but it seems to me like this would always return a projection result of type `update`? So it wouldn't be possible to perform a `delete` in a projection class?

Reply:
Ohh, yeah you're right. I'll fix that